runtime/hydradx/src/evm/runner.rs

```diff
@@ -158,7 +160,16 @@ where
 			weight_limit,
 			proof_size_base_cost,
 			config,
-		)
+		)?;
+
+		let CallInfo { exit_reason, .. } = &result;
+		let call_succeeded = matches!(
+			exit_reason,
+			ExitReason::Succeed(ExitSucceed::Returned) | ExitReason::Succeed(ExitSucceed::Stopped)
+		);
+		pallet_dispatcher::Pallet::<T>::set_last_evm_call_failed(!call_succeeded);
```

Comment:
why not just store `last_runner_result` and let whoever is reading that to deal with it?